src/browser_action/on-each-interaction.js

```diff
@@ -50,4 +90,36 @@ export function onEachInteraction(callback) {
     durationThreshold: 0,
     buffered: true,
   });
+
+  let recentLoAFs = [];
+
+  const getIntersectingLoAFs = (start, end) => {
+    const intersectingLoAFs = [];
+
+    for (let i = 0, loaf; (loaf = recentLoAFs[i]); i++) {
+      // If the LoAF ends before the given start time, ignore it.
+      if (loaf.startTime + loaf.duration < start) continue;
+
+      // If the LoAF starts after the given end time, ignore it and all
+      // subsequent pending LoAFs (because they're in time order).
+      if (loaf.startTime > end) break;
+
+      // Still here? If so this LoAF intersects with the interaction.
+      intersectingLoAFs.push(loaf);
+    }
+    return intersectingLoAFs;
+  };
+
+  const loafObserver = new PerformanceObserver((list) => {
+
+    for (const entry of list.getEntries()) { recentLoAFs.push(entry) };
```

Comment:
Nit: while probably not much faster, you could do this without the for-of loop:

```js
// Using `.concat()` (reasigns the variable)
recentLoAFs = recentLoAFs.concat(list.getEntries());

// or using `push()` (doesn't reassign the variable):
recentLoAFs.push(...list.getEntries());
```

Reply:
Good idea. Done.